### PR TITLE
Audio Support

### PR DESCRIPTION
**Pull request by Tomoli75**
Adds P:A support to CodeUtils, with a customisable backend URL.
Volume is customisable via changing the Noteblock / Jukebox sound level, or the master sound level.

### CODE REVIEW

File: src/main/java/io/github/codeutilities/audio/AudioHandler.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+public class AudioHandler {
+    String currentPlotId = "";
+    HashMap<String, HashSet<AudioClip>> tracks = new HashMap<>();
+    public AudioHandler() throws MalformedURLException {
+        try {
+            com.sun.javafx.application.PlatformImpl.startup(() -> {
+                URI uri = URI.create(ModConfig.getConfig().audioUrl);
+                String username = MinecraftClient.getInstance().getSession().getUsername();
+                IO.Options options = IO.Options.builder()
+                        .setQuery("username="+username+"&source="+("forge-"+CodeUtilities.MOD_ID+"-"+CodeUtilities.MOD_VERSION))
```
Comment (Reasonlesss) [inline src/main/java/io/github/codeutilities/audio/AudioHandler.java]: CodeUtilities is a fabric mod :)

File: src/main/java/io/github/codeutilities/config/ModConfig.java [hunk 2]
```diff
@@ -104,6 +104,10 @@
     @ConfigEntry.Gui.Tooltip()
     @ConfigEntry.BoundedDiscrete(min = 0, max = 100000)
     public int discordRPCTimeout = 15000;
+    @ConfigEntry.Category("misc")
+    @ConfigEntry.Gui.Tooltip()
+    @ConfigEntry.BoundedDiscrete(min = 0, max = 100000)
```
Comment (Reasonlesss) [inline src/main/java/io/github/codeutilities/config/ModConfig.java]: BoundedDiscrete is for numbers

**Comment by Reasonlesss** [comment]
Please also make this feature a toggle.

**Review by Tomoli75 — COMMENTED** [review]
Still attempting to get YouTube support working in Java, currently not functional

**Comment by Tomoli75** [comment]
Should now be all working! Ready for review

**Comment by ryandev2** [comment]
super cool. Reasonless is weak so he's sleeping rn but he'll review tomorrow i guess

**Review by Owen1212055 — CHANGES_REQUESTED** [review]
Please try to clear this large diff on the Lang file.

File: src/main/java/io/github/codeutilities/audio/AudioHandler.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                                    float volumeMaster = MinecraftClient.getInstance().options.getSoundVolume(SoundCategory.MASTER);
+                                    float volumeRecords = MinecraftClient.getInstance().options.getSoundVolume(SoundCategory.RECORDS);
+                                    float volume = volumeMaster * volumeRecords;
+                                    audio.setVolume(volume);
+                                }
+                            }
+                        }
+                    };
+                    ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+                    executor.scheduleAtFixedRate(handleVolume, 0, 1, TimeUnit.SECONDS);
```
Comment (Owen1212055) [inline src/main/java/io/github/codeutilities/audio/AudioHandler.java]: Store the executor on a private static final field.

**Review by Owen1212055 — CHANGES_REQUESTED** [review]

**Review by Owen1212055 — APPROVED** [review]
It's fine

**Pull request merged**